sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,233 @@ def _test_term(coeff, order):
+        # computing value of k
+        for i in _pow:
+            k = gcd(i, k)
+
+    # computing I0 of the given equation
+    I0 = powdenest(simplify(factor(((J1/k**2) - S(1)/4)/((x**k)**2))), force=True)
+    I0 = factor(simplify(powdenest(I0.subs(x, x**(S(1)/k)), force=True)))
+    num, dem = I0.as_numer_denom()
+
+    max_num_pow = max(_power_counting((num, )))
+    dem_args = dem.args
+    sing_point = []
+    dem_pow = []
+    # calculating singular point of I0.
+    for arg in dem_args:
+        if arg.has(x):
+            if isinstance(arg, Pow):
+                # (x-a)**n
+                dem_pow.append(arg.exp)
+                sing_point.append(solve(arg.base, x)[0])
+            else:
+                # (x-a) type
+                dem_pow.append(1)
+                sing_point.append(solve(arg, x)[0])
+
+    dem_pow.sort()
+    # checking if equivalance is exists or not.
+
+    if equivalance(max_num_pow, dem_pow) == "2F1":
+        return {'I0':I0, 'k':k, 'sing_point':sing_point, 'type':"2F1"}
+    else:
+        return {'I0':None, 'k':None, 'sing_point':None, 'type':None}
+
+def ode_2nd_hypergeometirc(eq, func, order, match):
+
+    from sympy.simplify.hyperexpand import hyperexpand
+    x = func.args[0]
+    f = func
+    C0, C1 = get_numbered_constants(eq, num=2)
+    a = match['a']
+    b = match['b']
+    c = match['c']
+    a3 = Wild('a3', exclude=[f, f.diff(x), f.diff(x, 2)])
+    b3 = Wild('b3', exclude=[f, f.diff(x), f.diff(x, 2)])
+    c3 = Wild('c3', exclude=[f, f.diff(x), f.diff(x, 2)])
+
+    A = cancel((match['r'][b3])/((match['r'])[a3]))
+    B = cancel((match['r'][c3])/((match['r'])[a3]))
+
+    sol = None
+    if match['type'] == "2F1":
+        if c.is_integer == False or c.is_positive is not True:
+            sol = C0*hyper([a, b], [c], x) + C1*hyper([a-c+1, b-c+1], [2-c], x)*x**(1-c)
+        elif c == 1:
+            y2 = Integral(exp(Integral((-(a+b+1)*x + c)/(x**2-x), x))/hyperexpand(hyper([a, b], [c], x)**2), x)*hyper([a, b], [c], x)
+            sol = C0*hyper([a, b], [c], x) + C1*y2
+        # applying transformation in the solution
+        subs = match['mobius']
+        dtdx = simplify(1/(subs.diff(x)))
+        _B = simplify(((a + b + 1)*x - c).subs(x, subs)*dtdx)
+        _B = simplify(_B + ((x**2 -x).subs(x, subs))*(dtdx.diff(x)*dtdx))
+        _A = simplify((x**2 - x).subs(x, subs)*(dtdx**2))
+        e = exp(logcombine(Integral(cancel(_B/(2*_A)), x).doit(), force=True))
+        sol = (sol).subs(x, match['mobius'])
+        sol = sol.subs(x, x**match['k'])
+        e = (e).subs(x, x**match['k'])
+
+        if B:
+            e1 = Integral(A/2, x).doit()
+            e1 = exp(logcombine(e1, force=True))
+
+        sol = cancel(powdenest(cancel(e/e1)*x**((-match['k']+1)/2), force=True))*(sol)
+        sol = Eq(func, sol)
+
+    # if sol is None then we can try for series solution
+    if sol is None:
+        try:
+            sol = dsolve(eq, func, hint = "2nd_power_series_regular", x0=x0)
```

Comment:
For now, Whenever we have match with 2F1 type then we can't say we will always going to get a solution as it depends upon the values of a, b,c also.
I have defined above `sol =  None`. If we will not get the solution from the `2F1` then we can try to find the solution from `2nd_power_series_regular`. 

But I can change the structure of functions so that we will only match with `2F1` whenever it is very clear that we will always get the solution from `2F1`

Reply:
I would like to remove the series solutions from `dsolve` ultimately so I think it's best we don't include them in more places.

We also need to restructure `dsolve` so that a hint can report a "possible match" but then fail later allowing another hint to proceed. We shouldn't embed that within particular hints though: it should be done at top-level in `dsolve`.